vms/avm/unique_tx.go

```diff
@@ -176,27 +173,23 @@ func (tx *UniqueTx) Dependencies() ([]snowstorm.Tx, error) {
 			continue
 		}
 		txID, _ := in.InputSource()
-		if txIDs.Contains(txID) {
-			continue
-		}
-		txIDs.Add(txID)
-		tx.deps = append(tx.deps, &UniqueTx{
+		inputTx := &UniqueTx{
 			vm:   tx.vm,
 			txID: txID,
-		})
+		}
+		if inputTx.Status() != choices.Accepted {
+			txIDs.Add(txID)
+		}
 	}
 	consumedIDs := tx.Tx.Unsigned.ConsumedAssetIDs()
 	for assetID := range tx.Tx.Unsigned.AssetIDs() {
-		if consumedIDs.Contains(assetID) || txIDs.Contains(assetID) {
+		if consumedIDs.Contains(assetID) {
 			continue
 		}
-		txIDs.Add(assetID)
-		tx.deps = append(tx.deps, &UniqueTx{
-			vm:   tx.vm,
-			txID: assetID,
-		})
+
+		return nil, errUnexpectedDependency
```

Comment:
After a mainnet + fuji sync passes with this then we can get rid of the whole loop + `ConsumedAssetIDs` check.

Reply:
they passed.